### PR TITLE
ramips: Add Xiaomi Mi Router 4A (100M International V2)

### PR DESCRIPTION
**Pull request by WereCatf**
The new V2 - version has a modified partition layout compared 
to the V1 and also has swapped the 5GHz chipset to the MT7613BE. 
This commit adds support for this new version of this router.

Signed-off-by: Nita Vesa <nita.vesa@elektrik.link>

### CODE REVIEW

**Comment by WereCatf** [comment]
I do not have the hardware so I have not been able to test this firmware myself, but so far one user reports having tested it and it working at https://forum.openwrt.org/t/xiaomi-4a-r4acv2-new-version-bricked/166933/12

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -154,7 +154,8 @@ wavlink,wl-wn578a2)
 	ucidef_set_led_switch "wan" "wan" "green:wan" "switch0" "0x10"
 	;;
 xiaomi,mi-router-4a-100m|\
-xiaomi,mi-router-4a-100m-intl)
+xiaomi,mi-router-4a-100m-intl\
```
Comment (obsy) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:157]: Missing |, should by | \

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -171,7 +171,8 @@ ramips_setup_interfaces()
 			"0:wan" "3:lan" "4:lan" "6@eth0"
 		;;
 	xiaomi,mi-router-4a-100m|\
-	xiaomi,mi-router-4a-100m-intl)
+	xiaomi,mi-router-4a-100m-intl\
```
Comment (obsy) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:174]: Missing |

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds (hunk 1 above)
Comment (WereCatf) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds]: Goddammit, I was certain I had those there. Well, fixed.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network (hunk 2 above)
Comment (WereCatf) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/02_network]: Fixed

File: target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts
Comment (AuthorReflex) [inline target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts:1]: if devices differ only in flash layout, then it is best practice to define a dtsi include file that containing common parts, and leave only device-specific changes in the device-specific dts file

File: target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&ethernet {
+	nvmem-cells = <&macaddr_factory_4>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <(-1)>;
```
Comment (rmilecki) [inline target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts:80]: Please don't add more `mac-address-increment` usages to OpenWrt

File: target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+};
+
+&factory {
+	compatible = "nvmem-cells";
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	macaddr_factory_4: macaddr@4 {
+		reg = <0x4 0x6>;
+	};
```
Comment (rmilecki) [inline target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts:95]: Please replace all `&factory` content with the following:
```
nvmem-layout {
	compatible = "fixed-layout";
	#address-cells = <1>;
	#size-cells = <1>;

	macaddr_factory_4: macaddr@4 {
		compatible = "mac-base";
		reg = <0x4 0x6>;
		#nvmem-cell-cells = <1>;
	};
};
```
then you can use `nvmem-cells = <&macaddr_factory_4 (-1)>;`

**Review by rmilecki — CHANGES_REQUESTED** [review]

**Comment by billic** [comment]
Since there is no activity for a long time I make own PR [14304](https://github.com/openwrt/openwrt/pull/14304) for same device. Please, review.

**Pull request closed without merging**